src/libraries/Microsoft.Extensions.Logging.Console/src/AnsiParser.cs

```diff
@@ -0,0 +1,201 @@
+
+using System;
+using System.IO;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    internal class AnsiParser
+    {
+        private readonly Action<string, int, int, ConsoleColor?, ConsoleColor?> _onParseWrite;
+        public AnsiParser(Action<string, int, int, ConsoleColor?, ConsoleColor?> onParseWrite)
+        {
+            if (onParseWrite == null)
+            {
+                throw new ArgumentNullException(nameof(onParseWrite));
+            }
+            _onParseWrite = onParseWrite;
+        }
+
+        /// <summary>
+        /// Parses a subset of display attributes
+        /// Set Display Attributes
+        /// Set Attribute Mode [{attr1};...;{attrn}m
+        /// Sets multiple display attribute settings. The following lists standard attributes that are getting parsed:
+        /// 1 Bright
+        /// Foreground Colours
+        /// 30 Black
+        /// 31 Red
+        /// 32 Green
+        /// 33 Yellow
+        /// 34 Blue
+        /// 35 Magenta
+        /// 36 Cyan
+        /// 37 White
+        /// Background Colours
+        /// 40 Black
+        /// 41 Red
+        /// 42 Green
+        /// 43 Yellow
+        /// 44 Blue
+        /// 45 Magenta
+        /// 46 Cyan
+        /// 47 White
+        /// </summary>
+        public void Parse(string message)
+        {
+            int startIndex = -1;
+            int length = 0;
+            ConsoleColor? foreground = null;
+            ConsoleColor? background = null;
+            var span = message.AsSpan();
+            const char EscapeChar = '\x1B';
+            ConsoleColor? color = null;
+            bool isBright = false;
+            for (int i = 0; i < span.Length; i++)
+            {
+                if (span[i] != EscapeChar || span.Length < i + 4 || span[i + 1] != '[')
+                {
+                    if (startIndex == -1)
+                    {
+                        startIndex = i;
+                    }
+                    int nextEscapeIndex = span.Slice(i, span.Length - i).IndexOf(EscapeChar);
+                    if (nextEscapeIndex == -1)
+                    {
+                        length += span.Length - i;
+                        break;
+                    }
+                    length += nextEscapeIndex;
+                    i += nextEscapeIndex - 1;
+                }
+                else if (span[i + 3] == 'm')
+                {
+#if NETSTANDARD2_0
+                    if (ushort.TryParse(span.Slice(i + 2, length: 1).ToString(), out ushort escapeCode))
+#else
+                    if (ushort.TryParse(span.Slice(i + 2, length: 1), out ushort escapeCode))
+#endif
+                    {
+                        if (escapeCode == 1)
+                            isBright = true;
+                        i += 3;
+                    }
+                }
+                else if (span.Length >= i + 5 && span[i + 4] == 'm')
+                {
+#if NETSTANDARD2_0
```

Comment:
Can we please change all these ifdefs that are specific for NETSTANDARD2_0 to also apply with netframework as this will break my current PR adding netfx configs for this?
```suggestion
#if (NETSTANDARD2_0 || NETFRAMEWORK)
```

Only adding it here but please apply this on all your ifdefs

Reply:
We should switch the `#if` around then - `#if NETCOREAPP` net core app code; `#else` non-net core app code;